application/src/ext/java/org/opentripplanner/ext/fares/impl/OrcaFareService.java

```diff
@@ -348,11 +361,22 @@ private Optional<Money> getSeniorFare(
     Leg leg
   ) {
     var route = leg.getRoute();
-    var regularFare = getRegularFare(fareType, rideType, defaultFare, leg);
+    if (route == null) {
+      return Optional.of(defaultFare);
+    }
     // Many agencies only provide senior discount if using ORCA
     return switch (rideType) {
-      case COMM_TRANS_LOCAL_SWIFT -> optionalUSD(1.25f);
-      case COMM_TRANS_COMMUTER_EXPRESS -> optionalUSD(2f);
+      case COMM_TRANS_LOCAL_SWIFT -> {
+        if (
+          leg
+            .getStartTime()
+            .isBefore(ZonedDateTime.of(2025, 3, 1, 0, 0, 0, 0, leg.getStartTime().getZone()))
```

Comment:
Extract a LocalDate and then call `atStartOfDay` with the leg's time zone as the argument.